### PR TITLE
Restore allowing fallback require from working directory

### PR DESCRIPTION
**Pull request by xg-wang**
This is the behavior for legacy whitelist.
Fix https://github.com/ember-fastboot/fastboot/issues/221 https://github.com/ember-fastboot/ember-cli-fastboot/issues/711

### CODE REVIEW

**Comment by xg-wang** [comment]
@rwjblue @kratiahuja @bobisjan Can you review?

File: src/ember-app.js [hunk 1]
```diff
@@ -136,8 +136,12 @@ class EmberApp {
       let isWhitelisted = whitelist.indexOf(packageName) > -1;
 
       if (isWhitelisted) {
-        let resolvedModulePath = resolve.sync(moduleName, { basedir: distPath });
-        return require(resolvedModulePath);
+        try {
+          let resolvedModulePath = resolve.sync(moduleName, { basedir: distPath });
+          return require(resolvedModulePath);
+        } catch (error) {
+          return require(moduleName);
```
Comment (snewcomer) [inline src/ember-app.js]: Is it fair to say this is _only_ applicable to the `development` case when `distPath` has no `node_modules`?

Comment (xg-wang) [reply src/ember-app.js]: Yes this is what this PR proposes. I asked the question about deployed case at https://github.com/ember-fastboot/fastboot/issues/198 and dist/node_modules is requried.

I updated the readme about deployment instructions.

Comment (xg-wang) [reply src/ember-app.js]: http://ember-fastboot.com/docs/user-guide#building-for-production should also be updated if PR is going to merge.

Comment (rwjblue) [inline src/ember-app.js]: We should tweak this to something akin to:

```suggestion
          if (error.code === 'MODULE_NOT_FOUND') {
            return require(moduleName);
          } else {
            throw error;
          }
```

**Comment by xg-wang** [comment]
@rwjblue @kratiahuja should this be released as a bug fix bump?

**Comment by gabrielcsapo** [comment]
bump on getting this merged 😄 🙏

**Comment by kratiahuja** [comment]
fastboot v2.0.2 is release with this fix only. Will release v3.0.0 later this week in order to ensure if there are other commits, they can go in.

**Comment by stefanpenner** [comment]
Thanks!

**Pull request merged**